app/api/routers/breeze_buddy/widget/handlers.py

```
@@ -414,80 +420,70 @@ async def update_widget_context_handler(
-
-        # Skip the write when nothing landed and there's no revision to
-        # record (e.g. template hasn't enabled the feature) — keeps inert
-        # pushes from churning the row.
-        if state_keys or facts_keys or body.revision is not None:
-            await upsert_agent_session_state(session_id, next_state)
+        )
+    except ClientContextTooLarge as exc:
+        raise HTTPException(
+            status_code=status.HTTP_413_REQUEST_ENTITY_TOO_LARGE,
+            detail=str(exc),
+        )
```

Ruff (0.15.17)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)